src/Sarif.Viewer.VisualStudio/ErrorList/SarifTableControlEventProcessorProvider.cs

```diff
@@ -44,42 +45,37 @@ private class EventProcessor : TableControlEventProcessorBase
             /// </summary>
             public override void PreprocessSelectionChanged(TableSelectionChangedEventArgs e)
             {
-                SarifErrorListItem sarifResult;
-                ListView errorList = (ListView)e.SelectionChangedEventArgs.Source;
-
-                if (errorList.SelectedItems.Count != 1)
+                // We only support single selection.
+                // So if there is no selection, or more than one, clear
+                // the SARIF explorer pane (set it's data context to null) and return.
+                ITableEntryHandle selectedTableEntry = e.AddedEntries.SingleOrDefault();
```

Comment:
>SingleOrDefault [](start = 70, length = 15)

The Enumerable implementation of this throws if it is not a singleton or empty.  Is that what happens here?

Reply:
+1 on this. `FirstOrDefault` would be safer (unless you want to find out the hard way on the day we start supporting multiple selection).

---
In reply to: [497147372](https://github.com/microsoft/sarif-visualstudio-extension/pull/206#discussion_r497147372) [](ancestors = 497147372)